### PR TITLE
Optimizing DDL Export Test Framework

### PR DESCRIPTION
**Pull request by hash-16**
### Description

Added two different ways of scripting the objects which can be switched by passing flag to the script according to the input file name.

### Issues Resolved

[BABEL-3969]

### Test Scenarios Covered ###
* **Use case based -**
Addition of DDL test files for the new test framework.

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

Signed-off-by: Ashish Prasad [pashisht@amazon.com](mailto:pashisht@amazon.com)

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/python/SMO_script.ps1 [hunk 1]
```diff
@@ -33,4 +41,42 @@ foreach ($CurrentObject in $Objects)
...
+{
+    $Objects = $db.Tables
+    $Objects += $db.Views
+    $Objects += $db.StoredProcedures
+    $Objects += $db.UserDefinedFunctions
+    $SubObjects += $db.Tables.Indexes
+    $SubObjects += $db.Tables.Triggers
+    foreach ($CurrentObject in $Objects)
+    {
+        if ($CurrentObject.schema -ne $schm -and $CurrentObject.schema -ne $dtb -and $CurrentObject.schema -ne $null -and -not $CurrentObject.IsSystemObject )
```
Comment (shalinilohia50) [inline test/python/SMO_script.ps1:57]: Can we remove this check now "-not $CurrentObject.IsSystemObject"?

File: test/python/SMO_script.ps1 [hunk 2]
```diff
@@ -33,4 +41,42 @@ foreach ($CurrentObject in $Objects)
...
+            $Scripter.Options.ScriptSchema = $True;
+            $Scripter.Options.ScriptData  = $False;
+            $Scripter.Options.NoCollation = $True;
+            $Scripter.Script($CurrentObject);
+            Write-Output "GO`n"
+        }
+    }
+    foreach ($CurrentObject in $SubObjects)
+    {
+        if (-not $CurrentObject.IsSystemObject )
```
Comment (shalinilohia50) [inline test/python/SMO_script.ps1:70]: Can we remove this check now "-not $CurrentObject.IsSystemObject"?

**Review by shalinilohia50 — APPROVED** [review]

File: test/python/SMO_script.ps1 (hunk 1 above)
Comment (hash-16) [reply test/python/SMO_script.ps1:57]: I removed the condition and tried but its actually taking the same amount of time as earlier and also returns some redundant GO in the output file. So i think it should not be removed.

File: test/python/SMO_script.ps1 (hunk 2 above)
Comment (hash-16) [reply test/python/SMO_script.ps1:70]: I removed the condition and tried but its actually taking the same amount of time as earlier and also returns some redundant GO in the output file. So i think it should not be removed.

**Review by shardgupta — APPROVED** [review]

**Pull request merged**